backend/scripts/github_collector.py

```
@@ -49,14 +49,24 @@ def __init__(self, repo_name: str = None, branch: str = None):
         self.branch = branch or "main"
         
         if not self.github_token:
-            raise ValueError("GITHUB_TOKEN not found in environment variables")
+            print("Warning: GITHUB_TOKEN not found in environment variables. Functionality will be limited to local cache.", file=sys.stderr)
+        
         if not self.github_repo:
             raise ValueError("GITHUB_REPO not found in environment variables")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)